src/test/java/ibm/jceplus/junit/openjceplusfips/TestRSASignatureWithSpecificSize.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright IBM Corp. 2024
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution.
+ */
+package ibm.jceplus.junit.openjceplusfips;
+
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.InvalidKeyException;
+
+import junit.framework.Test;
+import junit.framework.TestSuite;
+
+public class TestRSASignatureWithSpecificSize extends ibm.jceplus.junit.base.BaseTest {
+
+    //--------------------------------------------------------------------------
+    //
+    //
+    static final byte[] origMsg = "this is the original message to be signed I changed to a very long message to make sure enough bytes are there for copying."
+            .getBytes();
+
+    static {
+        Utils.loadProviderTestSuite();
+    }
+
+    //--------------------------------------------------------------------------
+    //
+    //
+    public TestRSASignatureWithSpecificSize() {
+        super(Utils.TEST_SUITE_PROVIDER_NAME);
+    }
+
+    // RSA signature sign allows at least 2048 bits of RSA key to be used for sign a signature.
+    public byte[] doSign(String sigAlgo, byte[] message, PrivateKey privateKey) throws Exception {
+        Signature sign = Signature.getInstance(sigAlgo, Utils.TEST_SUITE_PROVIDER_NAME);
+        try {
+            sign.initSign(privateKey);
+        } catch (InvalidKeyException ike) {
```

Comment:
The logic here and on line 66 seems to need to expect certain behavior/exceptions for certain key sizes. 

Can this be changed to expect the exception with message `In FIPS mode, only 2048, 3072, or 4096 size of RSA key is accepted.` only when testing key values other then 2048, 3072, or 4096? 

Similar for `RSA keys must be at least 2048 bits long` we only expect that failure on key lengths less then 2048.

Reply:
updated